[BUG] V7.2.0 Breaks Schema Validation Imports

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

Upgrading to version 7.2 produces errors when defining array parameters with pattern validation on the items

##### openapi-generator version

7.2

##### OpenAPI declaration file content or url
```
openapi: 3.0.3
paths:
  /api/v1/list:
    get:
      summary: List stuff
      description: |
        The description

      operationId: listStuff
      parameters:
        - $ref: "#/components/parameters/timestampQueryParam"
      responses:
        200:
          description: OK
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/StuffResponse"
      tags:
        - stuff

tags:
  - name: stuff
    description: somestuff
info:
  title: Example
  version: 0.0.0-SNAPSHOT
  license:
    name: Apache-2.0
    url: "https://www.apache.org/licenses/LICENSE-2.0.html"
  description: "description"

externalDocs:
  description: EXAMPLE
  url: "https://google.com"
components:
  schemas:
    StuffResponse:
      type: object
      properties:
        someProp:
          description: "This is a description"
          type: string
          example: "This is an example"
  parameters:
    timestampQueryParam:
      description: a timestamp param
      name: timestamp
      in: query
      explode: true
      schema:
        type: array
        items:
          type: string
          pattern: ^((eq|gt|gte|lt|lte|ne):)?\d{1,10}(.\d{1,9})?$
```

##### Generation Details

<!-- 
    Prefer CLI steps, including the language, libraries and various options. 
    Providing config-based settings allows for simpler testing across CLI and plugins. 
    For examples, see https://github.com/OpenAPITools/openapi-generator/tree/master/bin/configs
-->

##### Steps to reproduce

1. Configure the above spec and run `openApiGenerate`

##### Related issues/PRs

<!-- has a similar issue/PR been reported/opened before? Please do a search in https://github.com/openapitools/openapi-generator/issues?utf8=%E2%9C%93&q=is%3Aissue%20 -->

##### Suggest a fix

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit), or simply make a suggestion -->

The error produced with this spec is:
```
/StuffApi.java:91: error: cannot find symbol
  public StuffResponse listStuff(List<@Pattern(regexp = "^((eq|gt|gte|lt|lte|ne):)?\\d{1,10}(.\\d{1,9})?$")String> timestamp) throws ApiException {
                                       ^
  symbol:   class Pattern
  location: class StuffApi
/Users/jessenelson/IdeaProjects/hedera-mirror-node2/hedera-mirror-rest-java/build/generate-resources/main/com/hedera/mirror/rest/api/StuffApi.java:103: error: cannot find symbol
  public ApiResponse<StuffResponse> listStuffWithHttpInfo(List<@Pattern(regexp = "^((eq|gt|gte|lt|lte|ne):)?\\d{1,10}(.\\d{1,9})?$")String> timestamp) throws ApiException {
                                                                ^
  symbol:   class Pattern
  location: class StuffApi
/Users/jessenelson/IdeaProjects/hedera-mirror-node2/hedera-mirror-rest-java/build/generate-resources/main/com/hedera/mirror/rest/api/StuffApi.java:132: error: cannot find symbol
  private HttpRequest.Builder listStuffRequestBuilder(List<@Pattern(regexp = "^((eq|gt|gte|lt|lte|ne):)?\\d{1,10}(.\\d{1,9})?$")String> timestamp) throws ApiException {
                                                            ^
  symbol:   class Pattern
  location: class StuffApi
```

--- a/modules/openapi-generator/src/main/resources/Java/libraries/native/api.mustache
+++ b/modules/openapi-generator/src/main/resources/Java/libraries/native/api.mustache
@@ -13,6 +13,10 @@ import {{import}};
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
+{{#useBeanValidation}}
+import {{javaxPackage}}.validation.constraints.*;
+import {{javaxPackage}}.validation.Valid;
+{{/useBeanValidation}}
 {{#hasFormParamsInSpec}}
 import org.apache.http.HttpEntity;
 import org.apache.http.NameValuePair;

--- a/modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientCodegenTest.java
+++ b/modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientCodegenTest.java
@@ -615,6 +615,34 @@ public class JavaClientCodegenTest {
     }
 
     @Test
+    public void testJdkHttpClientWithUseBeanValidationEnabled() throws IOException {
+
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(CodegenConstants.API_PACKAGE, "xyz.abcdef.api");
+        properties.put(JavaClientCodegen.USE_BEANVALIDATION, true);
+        properties.put(JavaClientCodegen.USE_JAKARTA_EE, true);
+
+        File output = Files.createTempDirectory("test").toFile();
+        output.deleteOnExit();
+
+        final CodegenConfigurator configurator = new CodegenConfigurator()
+                .setGeneratorName("java")
+                .setLibrary(JavaClientCodegen.NATIVE)
+                .setAdditionalProperties(properties)
+                .setInputSpec("src/test/resources/3_1/issue-17485.yaml")
+                .setOutputDir(output.getAbsolutePath().replace("\\", "/"));
+
+        DefaultGenerator generator = new DefaultGenerator();
+        List<File> files = generator.opts(configurator.toClientOptInput()).generate();
+        files.forEach(File::deleteOnExit);
+
+        validateJavaSourceFiles(files);
+
+        TestUtils.assertFileContains(Paths.get(output + "/src/main/java/xyz/abcdef/api/UserApi.java"),
+                "@Pattern", "import jakarta.validation.constraints.*");
+    }
+
+    @Test
     public void testJdkHttpClientWithAndWithoutDiscriminator() throws Exception {
         Map<String, Object> properties = new HashMap<>();
         properties.put(CodegenConstants.API_PACKAGE, "xyz.abcdef.api");

--- a/modules/openapi-generator/src/test/resources/3_1/issue-17485.yaml
+++ b/modules/openapi-generator/src/test/resources/3_1/issue-17485.yaml
@@ -1,0 +1,25 @@
+# OAS document that tests for a regression when useBeanValidation=true and lists with a pattern.
+openapi: 3.1.0
+info:
+  title: issue-17485
+  version: 0.1.0
+servers:
+  - url: http://api.example.xyz/v1
+paths:
+  /user:
+    get:
+      tags:
+        - user
+      parameters:
+        - name: username
+          in: path
+          required: true
+          description: The name of the user
+          schema:
+            type: array
+            items:
+              type: string
+              pattern: "^[a-zA-Z0-9]$"
+      responses:
+        "200":
+          description: OK

